src/snowflake/cli/api/commands/execution_metadata.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) 2024 Snowflake Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import time
+from dataclasses import dataclass
+from enum import Enum
+
+
+class ExecutionStatus(Enum):
+    SUCCESS = "success"
+    FAILURE = "failure"
+
+
+@dataclass
+class ExecutionMetadata:
+    execution_id: str
```

Comment:
based on the usage we could initialise `execution_id` on creation like:
`execution_id: str = field(default_factory=lambda: uuid.uuid4().hex)`

or add a constructor to class:
```
@dataclass
class ExecutionMetadata:
    execution_id: str

    @classmethod
    def from_uuid():
         return cls(uuid.uuid4().hex)
```

and call it like  `ExecutionMetadata()` or `ExecutionMetadata.from_uuid()`